src/sage/rings/species.py

```diff
@@ -2505,3 +2506,59 @@ def factor(s, c, d):
                          for s in range(self._arity))
 
     Element = PolynomialSpeciesElement
+
+
+@cached_function
+def _atomic_set_like_species(n, names):
+    r"""
+    Return a list of the atomic set like species of degree `n`,
+    and provide their traditional names.
+
+    INPUT:
+
+        - ``n`` -- positive integer, the degree
+        - ``names`` -- an iterable of strings for the sorts of the
+          species
+
+    EXAMPLES::
+
+        sage: from sage.rings.species import _atomic_set_like_species
+        sage: _atomic_set_like_species(6, "X")
+        [E_2(E_3), E_2(X*E_2), E_2(X^3), E_3(E_2), E_3(X^2), E_6]
+
+        sage: l = [len(_atomic_set_like_species(n, "X")) for n in range(12)]
+        sage: l
+        [0, 1, 1, 1, 3, 1, 6, 1, 10, 4, 12, 1]
+        sage: oeis(l)                                                           # optional - internet
+        0: A007650: Number of set-like atomic species of degree n.
+
+        sage: _atomic_set_like_species(4, "U, V")
+        [E_2(E_2(V)), E_2(E_2(U)), E_2(V^2), E_2(U*V), E_2(U^2), E_4(U), E_4(V)]
+    """
+    if not n:
+        return []
+    M1 = MolecularSpecies("X")
+    M = MolecularSpecies(names)
+    if n == 1:
+        return [M(SymmetricGroup(1), {s: [1]}) for s in range(M._arity)]
```

Comment:
```suggestion
        return ()
    M1 = MolecularSpecies("X")
    M = MolecularSpecies(names)
    if n == 1:
        return tuple([M(SymmetricGroup(1), {s: [1]}) for s in range(M._arity)])
```
Just in case since it is a cached function.